File: examples/minimal.pl
use Glib::Object::Introspection;
Glib::Object::Introspection->setup(
    basename => 'Gtk',
    version => '3.0',
    package => 'Gtk');
Glib::Object::Introspection->setup(
    basename => 'GFlow',
    version => '0.2',
    package => 'GFlow');
Glib::Object::Introspection->setup(
    basename => 'GtkFlow',
    version => '0.2',
    package => 'GtkFlow');


Gtk::init(\@ARGV);

my $win = Gtk::Window->new('toplevel');
my $button = Gtk::Button->new_with_label("Foo");
my $vbox = Gtk::VBox->new('horizontal',0);
my $sw = Gtk::ScrolledWindow->new;
my $nodeview = GtkFlow::NodeView->new();
my $node, $wrap1, $wrap2, $source, $sink;
$button->signal_connect(clicked=>sub{
    my $n = create_node();
    print $n;
    $nodeview->add_node($n);
});

sub create_node {
    $node = new GFlow::SimpleNode;
    $wrap1 = Glib::Object::Introspection::GValueWrapper->new("Glib::Int", 0);
    $wrap2 = Glib::Object::Introspection::GValueWrapper->new("Glib::Int", 0);
    $source = GFlow::SimpleSource->new($wrap2);
    $source->set_name("source");
    $sink = GFlow::SimpleSink->new($wrap1);
    $sink->set_name("sink");

    $node->add_source($source);
    $node->add_sink($sink);
    $node->set_name("foo");
    return $node
}

$sw->add($nodeview);
$vbox->pack_start($button, 0, 0, 0);
$vbox->pack_start($sw, 1, 1, 1);
$win->add($vbox);
$win->show_all;
Gtk::main();
exit 0;

File: examples/minimal.lua
#! /usr/bin/env lua

# XXX:This example is outdated and uses libgtkflow in a pre-0.8-way

# TODO: fix this example. use the .with_type-constructor of Sources and Sinks


local lgi = require 'lgi'
local GObject = lgi.require('GObject')
local Gtk = lgi.require('Gtk')
local GFlow = lgi.require('GFlow')
local GtkFlow = lgi.require('GtkFlow')

local window = Gtk.Window {
   title = 'window',
   default_width = 400,
   default_height = 300,
   on_destroy = Gtk.main_quit
}

local sw = Gtk.ScrolledWindow {}
local nv = GtkFlow.NodeView.new()

local button =  Gtk.Button { label = 'Add Node' }
function button:on_clicked()
   local n = GFlow.SimpleNode {}
   local sink_v = GObject.Value(GObject.Type.INT,0)
   local source_v = GObject.Value(GObject.Type.INT,0)
   local sink = GFlow.SimpleSink.new(sink_v)
   local source = GFlow.SimpleSource.new(source_v)
   sink:set_name("sink")
   source:set_name("source")
   n:add_source(source)
   n:add_sink(sink)
   n:set_name("node")
   nv:add_node(n)
end

local vbox = Gtk.VBox()
sw:add(nv)
vbox:pack_start(button, false, false, 0)
vbox:pack_start(sw, true, true, 1)
window:add(vbox)

window:show_all()
Gtk.main() 
